src/main/webapp/js/common.js

```diff
@@ -581,12 +577,48 @@ function setStatusMessage(message, status) {
     
     $statusMessagesToUser.empty();
     $statusMessagesToUser.append($statusMessage);
-    $statusMessagesToUser.show();
+}
+
+/**
+ * Sets a status message and the message status.
+ * Default message type is info.
+ *
+ * @param message the text message to be shown to the user
+ * @param status type
+ */
+function setStatusMessage(message, status) {
+    if (message === '' || message === undefined || message === null) {
+        return;
+    }
+    prepareStatusMessage(message, status);
```

Comment:
As @damithc said, generally try to have stateless functions. I.e., you pass the function some arguments and it returns you everything you need, and not modify other things elsewhere.
https://github.com/TEAMMATES/teammates/pull/5753#discussion_r69849988

Right now, among other things, this `prepareStatusMessage()` creates a dependency on state changes external to the `setStatusMessage()` function.

Perhaps you can build the DOM entirely in jQuery for `setStatusMessageToForm()` after copying the DOM initially from the original statusMessageDiv